### DISCUSSION TITLE
Media library support Google Cloud Storage

### DISCUSSION THREAD

**Discussion by Vendin** [area17/twill#1025] [Q&A]
Does it already exist media library support Google Cloud Storage for main storage ?

What are the options ?

Is this planned at all ?

Thanks.

**Comment by ifox** [comment]
It does not exists, however because storage in Twill is based on Laravel Flysystem integration, as long as your Laravel application can access a Google Cloud Storage bucket, Twill should be able to use it. One challenge is our current frontend uploader component, which relies on direct upload from your browser to the storage provider, currently supporting S3 and Azure. It is planned in the future to move away from FineUploader, the library supporting that component, especially since FineUploader, sadly, is not maintained anymore (which is too bad because it was and is still to me the most complete client side upload solution).
